static/docs/command-reference/checkout.md

```diff
@@ -94,6 +83,13 @@ be pulled from remote storage using `dvc pull`.
   remove files that don't match those DVC-file references or are missing from
   cache. (They are not "committed", in DVC terms.)
 
+- `--relink` - ensures the file linking strategy (`reflink`, `hardlink`,
+  `symlink`, or `copy`) for all data files in the workspace is consistent with
+  the project's [`cache.type`](/doc/command-reference/config#cache). This is
+  achieved by restoring **all data files or a directories** referenced in
```

Comment:
probably restoring happens only if link type does not match, right?

Reply:
Good question. @Suor can you confirm please?

Reply:
(Seems like this PR has gone off Alex' radar so I just pinged him on Slack.)

Reply:
if `--relink` is specified then it always relinks, i.e. removes and links. There is no way in general case to check whether link type is correct. For example, we can't distinguish reflink from a copy. And even checking for hardlink is not completely reliable.

Reply:
Good to know! Leaving unresolved so it's easy to find for future reference.